modules/library/main/src/main/java/org/geotools/filter/function/FilterFunction_numberFormat.java

```diff
@@ -43,7 +43,6 @@
             parameter("format", String.class),
             parameter("number", Number.class));
     
-    public static DecimalFormatSymbols decimalFormatSymbols = DecimalFormatSymbols.getInstance(Locale.ENGLISH);
```

Comment:
There was a commit in the past dedicated to just set the English locale... it seems that you and @sampov2  have different needs there. If used locally for a test, the local locale seems to make more sense, if used inside a web application (e.g., GeoServer) using english seems to make more sense instead (e.g., GML is not using the default locale, it's using the English one). 
See: https://github.com/geotools/geotools/commit/d4a43a347ea81d70c1cdc5261a73186490997ce8

See also:
https://osgeo-org.atlassian.net/browse/GEOT-4987
https://osgeo-org.atlassian.net/browse/GEOT-4986

Time to add that third parameter to specify the intended locale?